### PR TITLE
Making StatusResponseHandler singleton and fixing all its instantiation invocations

### PR DESCRIPTION
**Pull request by sashidhar**
Making StatusResponseHandler singleton and fixing all its instantiation invocations.

### CODE REVIEW

**Comment by sashidhar** [comment]
@leventov ,   I see this failure.

`[ERROR] Failed to execute goal org.apache.maven.plugins:maven-remote-resources-plugin:1.5:process (process-resource-bundles) on project druid-hll: Error resolving project artifact: Could not transfer artifact org.hyperic:sigar:pom:1.6.5.132 from/to sonatype (https://oss.sonatype.org/content/repositories/releases/): Access denied to: https://oss.sonatype.org/content/repositories/releases/org/hyperic/sigar/1.6.5.132/sigar-1.6.5.132.pom, ReasonPhrase: Forbidden. for project org.hyperic:sigar:jar:1.6.5.132 -> [Help 1]`

**Comment by clintropolis** [comment]
>[ERROR] Failed to execute goal org.apache.maven.plugins:maven-remote-resources-plugin:1.5:process (process-resource-bundles) on project druid-hll: Error resolving project artifact: Could not transfer artifact org.hyperic:sigar:pom:1.6.5.132 from/to sonatype (https://oss.sonatype.org/content/repositories/releases/): Access denied to: https://oss.sonatype.org/content/repositories/releases/org/hyperic/sigar/1.6.5.132/sigar-1.6.5.132.pom, ReasonPhrase: Forbidden. for project org.hyperic:sigar:jar:1.6.5.132 -> [Help 1]

Travis was having some issues yesterday, this looks related to what I was seeing with other failures. I will restart and see if it does better today.

**Comment by sashidhar** [comment]
thanks @clintropolis 👍

File: core/src/main/java/org/apache/druid/java/util/http/client/response/StatusResponseHandler.java [hunk 1]
```diff
@@ -23,16 +23,24 @@
...
+import java.nio.charset.StandardCharsets;
 
 /**
  */
 public class StatusResponseHandler implements HttpResponseHandler<StatusResponseHolder, StatusResponseHolder>
 {
   private final Charset charset;
 
-  public StatusResponseHandler(Charset charset)
+  private static StatusResponseHandler statusResponseHandler = new StatusResponseHandler();
```
Comment (leventov) [inline core/src/main/java/org/apache/druid/java/util/http/client/response/StatusResponseHandler.java]: 1. Please make `final`
2. Conventionally, this constant is called `INSTANCE`

File: core/src/main/java/org/apache/druid/java/util/http/client/response/StatusResponseHandler.java [hunk 2]
```diff
@@ -23,16 +23,24 @@
 import org.jboss.netty.handler.codec.http.HttpResponse;
 
 import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
 
 /**
  */
 public class StatusResponseHandler implements HttpResponseHandler<StatusResponseHolder, StatusResponseHolder>
 {
   private final Charset charset;
```
Comment (leventov) [inline core/src/main/java/org/apache/druid/java/util/http/client/response/StatusResponseHandler.java]: Please remove this field.

File: indexing-service/src/main/java/org/apache/druid/indexing/overlord/hrtr/WorkerHolder.java [hunk 3]
```diff
@@ -71,7 +70,7 @@
   {
   };
 
-  private static final StatusResponseHandler RESPONSE_HANDLER = new StatusResponseHandler(StandardCharsets.UTF_8);
+  private static final StatusResponseHandler RESPONSE_HANDLER = StatusResponseHandler.getInstance();
```
Comment (leventov) [inline indexing-service/src/main/java/org/apache/druid/indexing/overlord/hrtr/WorkerHolder.java]: `StatusResponseHandler.getInstance()` can be used directly.

File: indexing-service/src/main/java/org/apache/druid/indexing/overlord/RemoteTaskRunner.java [hunk 4]
```diff
@@ -129,7 +128,7 @@
 public class RemoteTaskRunner implements WorkerTaskRunner, TaskLogStreamer
 {
   private static final EmittingLogger log = new EmittingLogger(RemoteTaskRunner.class);
-  private static final StatusResponseHandler RESPONSE_HANDLER = new StatusResponseHandler(StandardCharsets.UTF_8);
+  private static final StatusResponseHandler RESPONSE_HANDLER = StatusResponseHandler.getInstance();
```
Comment (leventov) [inline indexing-service/src/main/java/org/apache/druid/indexing/overlord/RemoteTaskRunner.java]: `StatusResponseHandler.getInstance()` can be used directly.

File: integration-tests/src/main/java/org/apache/druid/testing/clients/EventReceiverFirehoseTestClient.java [hunk 5]
```diff
@@ -61,7 +61,7 @@ public EventReceiverFirehoseTestClient(
   {
     this.host = host;
     this.jsonMapper = jsonMapper;
-    this.responseHandler = new StatusResponseHandler(StandardCharsets.UTF_8);
```
Comment (leventov) [inline integration-tests/src/main/java/org/apache/druid/testing/clients/EventReceiverFirehoseTestClient.java]: Please remove this field.

File: integration-tests/src/main/java/org/apache/druid/testing/clients/OverlordResourceTestClient.java [hunk 6]
```diff
@@ -64,7 +63,7 @@
     this.jsonMapper = jsonMapper;
     this.httpClient = httpClient;
     this.indexer = config.getIndexerUrl();
-    this.responseHandler = new StatusResponseHandler(StandardCharsets.UTF_8);
```
Comment (leventov) [inline integration-tests/src/main/java/org/apache/druid/testing/clients/OverlordResourceTestClient.java]: Please remove this field.

File: integration-tests/src/main/java/org/testng/DruidTestRunnerFactory.java [hunk 7]
```diff
@@ -119,7 +118,7 @@ private void runTests()
 
     public void waitUntilInstanceReady(final HttpClient client, final String host)
     {
-      final StatusResponseHandler handler = new StatusResponseHandler(StandardCharsets.UTF_8);
+      final StatusResponseHandler handler = StatusResponseHandler.getInstance();
```
Comment (leventov) [inline integration-tests/src/main/java/org/testng/DruidTestRunnerFactory.java]: This variable is unnecessary now.

File: integration-tests/src/test/java/org/apache/druid/tests/indexer/ITUnionQueryTest.java [hunk 8]
```diff
@@ -210,7 +209,7 @@ public void postEvents(int id) throws Exception
       LOG.info("Event Receiver Found at host [%s]", host);
 
       LOG.info("Checking worker /status/health for [%s]", host);
-      final StatusResponseHandler handler = new StatusResponseHandler(StandardCharsets.UTF_8);
+      final StatusResponseHandler handler = StatusResponseHandler.getInstance();
```
Comment (leventov) [inline integration-tests/src/test/java/org/apache/druid/tests/indexer/ITUnionQueryTest.java]: This variable is unnecessary now.

File: integration-tests/src/test/java/org/apache/druid/tests/security/ITBasicAuthConfigurationTest.java [hunk 9]
```diff
@@ -114,7 +113,7 @@
   @Client
   HttpClient httpClient;
 
-  StatusResponseHandler responseHandler = new StatusResponseHandler(StandardCharsets.UTF_8);
+  StatusResponseHandler responseHandler = StatusResponseHandler.getInstance();
```
Comment (leventov) [inline integration-tests/src/test/java/org/apache/druid/tests/security/ITBasicAuthConfigurationTest.java]: Please remove this field.

File: integration-tests/src/test/java/org/apache/druid/tests/security/ITTLSTest.java [hunk 10]
```diff
@@ -85,7 +84,7 @@
   @Inject
   TLSCertificateChecker certificateChecker;
 
-  StatusResponseHandler responseHandler = new StatusResponseHandler(StandardCharsets.UTF_8);
+  StatusResponseHandler responseHandler = StatusResponseHandler.getInstance();
```
Comment (leventov) [inline integration-tests/src/test/java/org/apache/druid/tests/security/ITTLSTest.java]: Please remove this field.

File: core/src/main/java/org/apache/druid/java/util/http/client/response/StatusResponseHandler.java (hunk 2 above)
Comment (sashidhar) [reply core/src/main/java/org/apache/druid/java/util/http/client/response/StatusResponseHandler.java]: @leventov , you mean to replace usage (each occurrence) of **charset** instance variable in this class directly with StandardCharsets.UTF_8

Comment (leventov) [reply core/src/main/java/org/apache/druid/java/util/http/client/response/StatusResponseHandler.java]: Yes

**Comment by sashidhar** [comment]
@leventov , implemented the review comments.

**Pull request merged**